app/src/main/java/com/example/concordia_campus_guide/Fragments/LocationFragment/LocationFragmentViewModel.java

```diff
@@ -280,4 +314,51 @@ public void setListOfPOI(@PoiType String poiType) {
         return poiList;
     }
 
+    public PolylineOptions getFloorPolylines(String floorCode) {
+        // previously drawindoorpaths
+        List<WalkingPoint> floorWalkingPoints = walkingPointsMap.get(floorCode);
+        PolylineOptions option = new PolylineOptions();
+        if(floorWalkingPoints == null) {
+            return option;
+        }
+        for(int i=0; i < floorWalkingPoints.size() - 1; i++) {
+            LatLng point1 = floorWalkingPoints.get(i).getCoordinate().getLatLng();
+            LatLng point2 = floorWalkingPoints.get(i + 1).getCoordinate().getLatLng();
+            option.add(point1, point2);
+        }
+        return option
+                .width(10)
+                .pattern(ClassConstants.WALK_PATTERN)
+                .color(Color.rgb(147,35, 57))
+                .visible(true);
+    }
+
+    public void drawOutdoorPath(List<DirectionWrapper> outdoorDirections, GoogleMap map) {
+        for(DirectionWrapper directionWrapper: outdoorDirections) {
+            PolylineOptions polylineOptions = stylePolyLine(directionWrapper.getDirection().getTransportType());
+            List<com.example.concordia_campus_guide.GoogleMapsServicesTools.GoogleMapsServicesModels.LatLng> polyline = directionWrapper.getPolyline().decodePath();
+
+            for(int i=0; i<polyline.size(); i++){
+                polylineOptions.add(new LatLng(polyline.get(i).lat, polyline.get(i).lng));
+            }
+            map.addPolyline(polylineOptions);
+        }
+    }
+
+    private PolylineOptions stylePolyLine(String type) {
+        PolylineOptions polylineOptions = new PolylineOptions().width(20);
+        if(type.equals(ClassConstants.WALKING)) {
```

Comment:
Same comment as earlier, we can just use the enum directly and use the method getName().